### ISSUE TITLE
Destination-elasticsearch really implementing the full_refresh - overwrite sync?

### ISSUE THREAD

**Issue by marcosmarxm** [airbytehq/airbyte#16182]
This Github issue is synchronized with Zendesk:

**Ticket ID:** [#2084](https://airbyte7538.zendesk.com/agent/tickets/2084)
**Priority:** normal
**Group:** Community Assistance Engineer
**Assignee:** Marcos Marx

**Original ticket description:**

<blockquote><div class="zd-comment" dir="auto"><p dir="auto">I’m experimenting between the different syncModes and I would have imagined if I ingested 100 records into Elasticsearch and I had the sync set to full_refresh - overwrite, then when I ran the sync again it would delete the index and ingest the same 100 records? However, I get 200 records in the index.</p>
<p dir="auto">I’m running version 0.1.3.</p>
<p dir="auto">Taking a quick look through the code (Maybe around here: <a href="https://github.com/airbytehq/airbyte/blob/09aa685aadbb6dc45f66b2c87d4c13af9b88e6b2/airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchAirbyteMessageConsumerFactory.java#L122" rel="noopener nofollow ugc noreferrer" style="color: #006699; font-weight: bold; text-decoration: none;">airbyte/ElasticsearchAirbyteMessageConsumerFactory.java at 09aa685aadbb6dc45f66b2c87d4c13af9b88e6b2 · airbytehq/airbyte · GitHub</a>), I didn’t see it delete the index based on syncMode.</p>
<p dir="auto">Wanted to check with people that might know more because the docs say that the full_refresh syncMode is supported.</p> 

[<a href="https://discuss.airbyte.io/t/destination-elasticsearch-really-implementing-the-full-refresh-overwrite-sync/2396/1" rel="noreferrer">Discourse post</a>]</div></blockquote>

**Comment by marcosmarxm** [comment]
*Comment made from Zendesk by Marcos Marx on 2022-08-31 at 18:39:*

<blockquote><div class="zd-comment" dir="auto">Thanks for reporting this Ryan I opened the issue <a rel="noopener noreferrer" href="https://github.com/airbytehq/airbyte/issues/16182">https://github.com/airbytehq/airbyte/issues/16182</a> to solve the issue.<br></div></blockquote>

**Comment by wirelessrpm** [comment]
I think I have a fix for this written on a fork I made of the project. I'm probably looking right at it, but are there instructions on how you'd like me to make the PR for it?

**Comment by wirelessrpm** [comment]
I've written a fix for this on my fork, but not sure where to aim it when submitting the PR.

**Comment by marcosmarxm** [comment]
@wirelessrpm please read Airbyte's docs about contributing to the project: https://docs.airbyte.com/contributing-to-airbyte/

**Comment by joelluijmes** [comment]
FYI: I just also ran into this. I dug a bit in the code, and the `OVERWRITE` mode only works when the Upsert Records checkbox is deselected (default it is on). When it is off, it performs a sync to a temporary index which replaces the original index. I suppose to have some form of atomic replacement to mitigate any downtime.

- [ElasticsearchWriteConfig.java#L86-L88](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchWriteConfig.java#L86-L88) determines whether to use temporary index.
- [ElasticsearchAirbyteMessageConsumerFactory.java#L125-L131](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchAirbyteMessageConsumerFactory.java#L125-L131) selects which index to write to
- [ElasticsearchAirbyteMessageConsumerFactory.java#L68-L70](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchAirbyteMessageConsumerFactory.java#L68-L70) finally replaces the index

Not sure if this is the most obvious solution, but it does seem to work 👍

**Comment by khiem20tc** [comment]
> FYI: I just also ran into this. I dug a bit in the code, and the `OVERWRITE` mode only works when the `Upsert Records` checkbox is deselected (default it is on). When it is off, it performs a sync to a temporary index which replaces the original index. I suppose to have some form of atomic replacement to mitigate any downtime.
> 
> * [ElasticsearchWriteConfig.java#L86-L88](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchWriteConfig.java#L86-L88) determines whether to use temporary index.
> * [ElasticsearchAirbyteMessageConsumerFactory.java#L125-L131](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchAirbyteMessageConsumerFactory.java#L125-L131) selects which index to write to
> * [ElasticsearchAirbyteMessageConsumerFactory.java#L68-L70](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-elasticsearch/src/main/java/io/airbyte/integrations/destination/elasticsearch/ElasticsearchAirbyteMessageConsumerFactory.java#L68-L70) finally replaces the index
> 
> Not sure if this is the most obvious solution, but it does seem to work 👍
> 
> TLDR: can be fixed by disabling the `Upsert Records` config.

Hi sir, how to disable *Upsert Records* to use Overwrite mode with ElasticSearch? Where can I can config in Airbyte source code or in ElasticSearch config?

**Comment by joelluijmes** [comment]
’Upsert Records’ is just a configuration setting in Airbyte when setting up the connection. You don't need to change the code.

**Comment by khiem20tc** [comment]
Hi Sir, I don't see that config during the setup of my connections. Plz show me where can I find that configuration? Many thanks!

**Comment by octavia-squidington-iii** [comment]
At Airbyte, we seek to be clear about the project priorities and roadmap. This issue has not had any activity for 365 days, suggesting that it's not as critical as others. It's possible it has already been fixed. It is being marked as stale and will be closed in 20 days if there is no activity. To keep it open, please comment to let us know why it is important to you and if it is still reproducible on recent versions of Airbyte.
